x-pack/heartbeat/scenarios/framework/fakeloader.go

```diff
@@ -27,29 +29,33 @@ func newLoaderDB() *loaderDB {
 	}
 }
 
-func (ldb loaderDB) AddState(sf stdfields.StdMonitorFields, state *monitorstate.State) {
+func (ldb *loaderDB) AddState(sf stdfields.StdMonitorFields, state *monitorstate.State) {
 	ldb.mtx.Lock()
 	defer ldb.mtx.Unlock()
 
-	ldb.lastTime = time.Now()
-
-	ldb.keysToState[monitorstate.LoaderDBKey(sf, ldb.lastTime, 0)] = state
+	key := keyFor(sf)
```

Comment:
The previous behavior here was broken, but in a way that didn't matter for previous tests. Basically, the previous state would not be read unless the timestamps matched, which worked when all data was unix epoch. That's not the actual behavior of a simulated ES. We should simply load the last state. This was a misuse of the `LoaderDBKey` which was only ever meant to be used to append data, not to deterministically query old data.